From e0e68693414d123275ef1f34dfd7a3b5da4bba6d Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Mon, 24 Oct 2011 12:32:50 +0200
Subject: [PATCH 08/16] Reading 26 item in array with size 12.

Walking through array chconfig (of size only 12 items)
by for loop with stop-value 26.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 lsusb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lsusb.c b/lsusb.c
index 73a1ef4..2ccd947 100644
--- a/lsusb.c
+++ b/lsusb.c
@@ -1463,7 +1463,7 @@ static void dump_audiocontrol_interface(libusb_device_handle *dev, const unsigne
 				printf("        baSourceID(%2u)      %5u\n", i, buf[7+i]);
 			printf("        bNrChannels         %5u\n"
 			       "        wChannelConfig      %5u\n", buf[7+j], chcfg);
-			for (i = 0; i < 26; i++)
+			for (i = 0; i < 12; i++)
 				if ((chcfg >> i) & 1)
 					printf("          %s\n", chconfig[i]);
 			printf("        iChannelNames       %5u %s\n"
-- 
2.4.3

